cmake/MakeChm.cmake

```diff
@@ -66,4 +66,4 @@ execute_process(
     ERROR_QUIET
 )
 
-file(COPY "${htmlhelpdir}/${docname}.chm" DESTINATION "${BUILD_DIR}")
+execute_process(COMMAND ${CMAKE_COMMAND} -E copy "${htmlhelpdir}/${docname}.chm" "${BUILD_DIR}")
```

Comment:
There are some reasons.

* I'm not familiar with cmake and ninja. 
* In other locations, execute_process is used for file copy and directory copy. Therefore I'd like to unify them.
* I prefer "cp src dest" format. And I don't like "cp src DESTINATION dest" format.

Would you tell me good manners and commands to copy files and directories in the cmake?
I'd like to follow the best practice of cmake.

Reply:
I think `file(COPY ....)` is preferred over `execute_process` at least in this  case.

`file(COPY ...)` is a built-in command, executed in-process by cmake.  IIRC it also tracks file changes and will automatically update copied files in case any of the source files changes.

`execute_process` on the other hand will tell cmake to run an external command. In this  case that command happens to be cmake again, but this  case in script mode to call the builtin script command `copy`. Aside from the unwanted recursion, the main disadvantage of this mode is that the calling cmake command doesn't know anything of the changes made by the called command, so we lose tracking of source file changes. The script mode of cmake is to be used when cmake commands should be called at build time (that is while make/ninja is running) rather than during cmake's configuration or generation phases or as part of custom commands if not other option is available.

In retrospect this probably goes for all the places where you changed `file(COPY ...)` to a `COMMAND cmake -E copy` statement.

Reply:
OK. I will look up the cmake manual and unify the procedures.

@gjanssens 
On another topic, why are the files split into MakeChm.cmake and AddChmTraget.cmake for chm?
I'd like to merge them into AddChmTraget.cmake. Do you have some technical problems?

Reply:
I cleaned up AddChmTraget.cmake and removed MakeChm.cmake.